pkg/operator2/transport.go

```diff
@@ -22,21 +22,27 @@ func transportFor(serverName string, caData, certData, keyData []byte) (http.Rou
 }
 
 func transportForInner(serverName string, caData, certData, keyData []byte) (http.RoundTripper, error) {
-	if len(caData) == 0 && len(certData) == 0 && len(keyData) == 0 {
-		return http.DefaultTransport, nil
-	}
-
-	if (len(certData) == 0) != (len(keyData) == 0) {
-		return nil, errors.New("cert and key data must be specified together")
-	}
-
 	// copy default transport
 	transport := net.SetTransportDefaults(&http.Transport{
 		TLSClientConfig: &tls.Config{
 			ServerName: serverName,
 		},
 	})
 
+	if len(caData) == 0 && len(certData) == 0 && len(keyData) == 0 {
+		// reload system cert pool in case trusted-ca-bundle changed
+		systemCertPool, err := x509.SystemCertPool()
```

Comment:
`SystemCertPool` does not reload the system roots.  After the initial run the result is cached.

Reply:
removed that code, should have checked more thoroughly, thanks